### PR TITLE
INTLY-7238 Switched from update to patch

### PR DESCRIPTION
**Pull request by CathalOConnorRH**
Switched from update to patch and check for readyreplicas before passing test.

# Description
Switched from update to patch and check for readyreplicas before passing test.
Tests were reported as failing in these two tickets. 
- https://issues.redhat.com/browse/INTLY-7238 
- https://issues.redhat.com/browse/INTLY-7556

## Type of change

- [X ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added a test case that will be used to verify my changes 
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

File: test/common/scale_up_replicas_apicurito.go [hunk 1]
```diff
@@ -46,16 +48,52 @@ func TestReplicasInApicurito(t *testing.T, ctx *TestingContext) {
 		t.Fatalf("Incorrect number of replicas : %v", err)
 	}
 
-	apicuritoCR, err = updateApicurito(ctx.Client, scaleDownApicuritoReplicas, t)
+	t.Log("Chekcing replicas are ready")
```
Comment (pawelpaszki) [inline test/common/scale_up_replicas_apicurito.go:51]:
```suggestion
	t.Log("Checking replicas are ready")
```

File: test/common/scale_up_replicas_apicurito.go [hunk 2]
```diff
@@ -46,16 +48,52 @@ func TestReplicasInApicurito(t *testing.T, ctx *TestingContext) {
...
+	if err := checkReplicasAreReady(ctx, t, scaleUpApicuritoReplicas); err != nil {
+		t.Fatalf("Replicas weren't ready within timeout")
+	}
+
+	apicuritoCR, err = updateApicurito(ctx, scaleDownApicuritoReplicas, t)
 	if err != nil {
 		t.Fatalf("Unable to update : %v", err)
 	}
 
+	t.Log("Chekcing replicas are ready")
```
Comment (pawelpaszki) [inline test/common/scale_up_replicas_apicurito.go:61]:
```suggestion
	t.Log("Checking replicas are ready")
```

File: test/common/scale_up_replicas_apicurito.go [hunk 3]
```diff
@@ -46,16 +48,52 @@ func TestReplicasInApicurito(t *testing.T, ctx *TestingContext) {
...
+	if err := checkReplicasAreReady(ctx, t, scaleDownApicuritoReplicas); err != nil {
+		t.Fatalf("Replicas weren't ready within timeout")
+	}
+
 	t.Log("Checking correct number of updated replicas are set")
 	if err := checkNumberOfReplicasAgainstValueApicurito(apicuritoCR, ctx, numberOfApicuritoReplicas, retryIntervalApicurito, timeoutApicurito, t); err != nil {
 		t.Fatalf("Incorrect number of replicas : %v", err)
 	}
 
+	t.Log("Chekcing replicas are ready")
```
Comment (pawelpaszki) [inline test/common/scale_up_replicas_apicurito.go:71]:
```suggestion
	t.Log("Checking replicas are ready")
```

File: test/common/scale_up_replicas_rhsso.go [hunk 4]
```diff
@@ -38,7 +41,12 @@ func checkScalingOfKeycloakReplicas(t *testing.T, ctx *TestingContext, keycloakC
 		t.Fatalf("Incorrect number of replicas to start : %v", err)
 	}
 
-	keycloakCR, err = updateKeycloakCR(ctx.Client, scaleUpRhssoReplicas, keycloakCRName, keycloakCRNamespace)
+	t.Log("Chekcing replicas are ready")
```
Comment (pawelpaszki) [inline test/common/scale_up_replicas_rhsso.go:44]:
```suggestion
	t.Log("Checking replicas are ready")
```

File: test/common/scale_up_replicas_rhsso.go [hunk 5]
```diff
@@ -48,17 +56,27 @@ func checkScalingOfKeycloakReplicas(t *testing.T, ctx *TestingContext, keycloakC
 		t.Fatalf("Incorrect number of replicas : %v", err)
 	}
 
-	keycloakCR, err = updateKeycloakCR(ctx.Client, scaleDownRhssoReplicas, keycloakCRName, keycloakCRNamespace)
+	t.Log("Chekcing replicas are ready")
```
Comment (pawelpaszki) [inline test/common/scale_up_replicas_rhsso.go:59]:
```suggestion
	t.Log("Checking replicas are ready")
```

File: test/common/scale_up_replicas_rhsso.go [hunk 6]
```diff
@@ -67,6 +85,11 @@ func checkScalingOfKeycloakReplicas(t *testing.T, ctx *TestingContext, keycloakC
 	if err := checkNumberOfReplicasAgainstValueRhsso(keycloakCR, ctx, numberOfRhssoReplicas, retryInterval, timeout, t); err != nil {
 		t.Fatalf("Incorrect number of replicas : %v", err)
 	}
+
+	t.Log("Chekcing replicas are ready")
```
Comment (pawelpaszki) [inline test/common/scale_up_replicas_rhsso.go:89]:
```suggestion
	t.Log("Checking replicas are ready")
```

Comment (CathalOConnorRH) [reply test/common/scale_up_replicas_rhsso.go:89]: Thanks for those. will push a new version now.

**Comment by pawelpaszki** [comment]
ran the tests locally, but would be interested to see what @davidffrench has to say.

File: test/common/scale_up_replicas_apicurito.go [hunk 7]
```diff
@@ -46,16 +48,52 @@ func TestReplicasInApicurito(t *testing.T, ctx *TestingContext) {
...
 
+	t.Log("Checking replicas are ready")
+	if err := checkReplicasAreReady(ctx, t, numberOfApicuritoReplicas); err != nil {
+		t.Fatalf("Replicas weren't ready within timeout")
+	}
+
+}
+
+func checkReplicasAreReady(dynClient *TestingContext, t *testing.T, replicas int32) error {
+	deployment, err := dynClient.KubeClient.AppsV1().Deployments(apicuritoNamespace).Get(apicuritoName, metav1.GetOptions{})
```
Comment (davidffrench) [inline test/common/scale_up_replicas_apicurito.go:79]: This should not be required with the same get inside the poll on line 83 https://github.com/integr8ly/integreatly-operator/pull/754/files#diff-c390ece00887543e97f361f7e04d648aR83

File: test/common/scale_up_replicas_apicurito.go [hunk 8]
```diff
@@ -68,26 +106,27 @@ func getApicurito(dynClient k8sclient.Client) (apicuritov1alpha1.Apicurito, erro
 	return *apicuritoCR, nil
 }
 
-func updateApicurito(dynClient k8sclient.Client, replicas int32, t *testing.T) (apicuritov1alpha1.Apicurito, error) {
+func updateApicurito(dynClient *TestingContext, replicas int32, t *testing.T) (apicuritov1alpha1.Apicurito, error) {
```
Comment (davidffrench) [inline test/common/scale_up_replicas_apicurito.go:109]:
```suggestion
func updateApicurito(ctx *TestingContext, replicas int32, t *testing.T) (apicuritov1alpha1.Apicurito, error) {
```

Small name change but keep it consistent with other functions.

**Comment by CathalOConnorRH** [comment]
> Good work @CathalOConnorRH .
> 
> Could you tell me again what was happening when the deployment was not in a ready state?

When this test completes a subsequent test fails due to the pods not being ready when the next tests tries to access them. By adding in the check here we try to ensure it's safe to proceed when the pods are ready.

**Comment by davidffrench** [comment]
> > Good work @CathalOConnorRH .
> > Could you tell me again what was happening when the deployment was not in a ready state?
> 
> When this test completes a subsequent test fails due to the pods not being ready when the next tests tries to access them. By adding in the check here we try to ensure it's safe to proceed when the pods are ready.

Perfect, thank you. Do you think it is worth just adding the ready check at the end of the tests?

**Comment by CathalOConnorRH** [comment]
> > > Good work @CathalOConnorRH .
> > > Could you tell me again what was happening when the deployment was not in a ready state?
> > 
> > 
> > When this test completes a subsequent test fails due to the pods not being ready when the next tests tries to access them. By adding in the check here we try to ensure it's safe to proceed when the pods are ready.
> 
> Perfect, thank you. Do you think it is worth just adding the ready check at the end of the tests? This might reduce the time it is taking to run

That depends. are we happy to assume that the scaling up is ok regardless of the pod status once the CR is updated ? If we are then just checking the pods are ready on the last reset test should be ok.

**Comment by davidffrench** [comment]
> > > > Good work @CathalOConnorRH .
> > > > Could you tell me again what was happening when the deployment was not in a ready state?
> > > 
> > > 
> > > When this test completes a subsequent test fails due to the pods not being ready when the next tests tries to access them. By adding in the check here we try to ensure it's safe to proceed when the pods are ready.
> > 
> > 
> > Perfect, thank you. Do you think it is worth just adding the ready check at the end of the tests? This might reduce the time it is taking to run
> 
> That depends. are we happy to assume that the scaling up is ok regardless of the pod status once the CR is updated ? If we are then just checking the pods are ready on the last reset test should be ok.

I would say yes, this would then be testing that 3sacle itself for example is correctly updating the deployment and testing the deployment is ready but this test should just be that we can modify the replicas in the CR

**Comment by CathalOConnorRH** [comment]
> > > > > Good work @CathalOConnorRH .
> > > > > Could you tell me again what was happening when the deployment was not in a ready state?
> > > > 
> > > > 
> > > > When this test completes a subsequent test fails due to the pods not being ready when the next tests tries to access them. By adding in the check here we try to ensure it's safe to proceed when the pods are ready.
> > > 
> > > 
> > > Perfect, thank you. Do you think it is worth just adding the ready check at the end of the tests? This might reduce the time it is taking to run
> > 
> > 
> > That depends. are we happy to assume that the scaling up is ok regardless of the pod status once the CR is updated ? If we are then just checking the pods are ready on the last reset test should be ok.
> 
> I would say yes, this would then be testing that 3sacle itself for example is correctly updating the deployment and testing the deployment is ready but this test should just be that we can modify the replicas in the CR

ok perfect, I'll update the tests to only check the pods at the last step.

**Comment by matskiv** [comment]
@davidffrench @pawelpaszki Hey guys, you reviewed this already, can you give final look and approve it all looks good? Thank you.

**Pull request merged**